x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/textembedding/MultilingualE5SmallInternalServiceSettings.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * this file was contributed to by a generative AI
+ */
+
+package org.elasticsearch.xpack.inference.services.textembedding;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+import org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.elasticsearch.TransportVersions.ML_TEXT_EMBEDDING_INFERENCE_SERVICE_ADDED;
+
+public class MultilingualE5SmallInternalServiceSettings extends TextEmbeddingInternalServiceSettings {
+
+    public static final String NAME = "multilingual_e5_small_service_settings";
+
+    public MultilingualE5SmallInternalServiceSettings(int numAllocations, int numThreads, String modelId) {
+        super(numAllocations, numThreads, modelId);
+    }
+
+    public MultilingualE5SmallInternalServiceSettings(StreamInput in) throws IOException {
+        super(in.readVInt(), in.readVInt(), in.readString());
+    }
+
+    /**
+     * Parse the MultilingualE5SmallServiceSettings from map and validate the setting values.
+     *
+     * If required setting are missing or the values are invalid an
+     * {@link ValidationException} is thrown.
+     *
+     * @param map Source map containing the config
+     * @return The {@code MultilingualE5SmallServiceSettings} builder
+     */
+    public static MultilingualE5SmallInternalServiceSettings.Builder fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+        Integer numAllocations = ServiceUtils.removeAsType(map, NUM_ALLOCATIONS, Integer.class);
+        Integer numThreads = ServiceUtils.removeAsType(map, NUM_THREADS, Integer.class);
+
+        validateParameters(numAllocations, validationException, numThreads);
+
+        String modelId = ServiceUtils.removeAsType(map, MODEL_ID, String.class);
+        if (modelId != null) {
```

Comment:
Just double checking that we allow `modelId` to be null?

Reply:
Currently, for built-in internal services, we do allow modelId to be null as we automatically select the best model for the platform. I guess its an open question, whether or not this is the best approach.

Reply:
I just realized that because of the logic flow from the TextEmbeddingIntenalService it isn't even currently possible for modelId to be null here. However, if that changed in the future then my above comment would apply.